internal/oc/release_test.go

```diff
@@ -605,6 +605,34 @@ var _ = Describe("getIcspFileFromRegistriesConfig", func() {
 	})
 })
 
+var _ = Describe("GetReleaseBinaryPath", func() {
+	var r Release
+	BeforeEach(func() {
+		r = NewRelease(nil, Config{}, nil)
+	})
+
+	It("returns the openshift-baremetal-install binary for versions earlier than 4.16", func() {
+		_, bin, _, err := r.GetReleaseBinaryPath("image", "dir", "4.15.0")
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(bin).To(Equal("openshift-baremetal-install"))
+	})
+
+	It("returns the openshift-install binary for 4.16.0", func() {
+		_, bin, _, err := r.GetReleaseBinaryPath("image", "dir", "4.16.0")
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(bin).To(Equal("openshift-install"))
+	})
+
+	It("returns the openshift-install binary for 4.16 pre release", func() {
+		_, bin, _, err := r.GetReleaseBinaryPath("image", "dir", "4.16.0-ec.6")
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(bin).To(Equal("openshift-install"))
+		_, bin, _, err = r.GetReleaseBinaryPath("image", "dir", "4.16.0-rc.0")
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(bin).To(Equal("openshift-install"))
+	})
```

Comment:
Should we future-proof this by adding a test case for 4.17 as well?

Reply:
Done